apps/src/code-studio/pd/application/teacher1920/Teacher1920Application.jsx

```diff
@@ -93,4 +94,95 @@ export default class Teacher1920Application extends FormController {
     ga('set', 'page', url);
     ga('send', 'pageview');
   }
+
+  /**
+   * @override
+   */
+  setPage(i) {
+    const newPage = Math.min(
+      Math.max(i, 0),
+      this.getPageComponents().length - 1
+    );
+
+    this.setState({
+      currentPage: newPage
+    });
+
+    if (this.triedToSubmit) {
+      let pageWithErrors = this.validateForm();
+
+      // If errors exist, create a summary header containing
+      // clickable links to pages that have errors.
+      if (pageWithErrors.length) {
+        this.setState({
+          globalError: true,
+          errorHeader: (
+            <span>
+              Please fill out all required fields on page
+              {pageWithErrors.length > 1 ? 's' : ''}{' '}
+              {pageWithErrors.map(index => (
+                <a
+                  key={index}
+                  onClick={() => this.setPage(index)}
+                  style={{cursor: 'pointer'}}
+                >
+                  {' '}
+                  {index + 1}
+                </a>
+              ))}
+              . Once you are done, head to the last page to confirm and submit
+              your application.
+            </span>
```

Comment:
We can clean this up a bit by extracting a pure functional component.  Here:
```jsx
this.setState({
  globalError: true,
  errorHeader: <InvalidPagesSummary pages={pagesWithErrors} setPage={this.setPage.bind(this)}/>
});
```

And later, or even in a separate file:
```jsx
const InvalidPagesSummary = ({pages, setPage}) => (
  <span>
    Please fill out all required fields on {pages.length > 1 ? 'pages' : 'page'}{' '}
    {pages
      .map(p => (
        <a key={p} onClick={() => setPage(p)} style={{cursor: 'pointer'}}>
          {p + 1}
        </a>
      ))
      .reduce((prev, curr) => [prev, ', ', curr])}
    . Once you are done, head to the last page to confirm and submit your
    application.
  </span>
);
InvalidPagesSummary.propTypes = {
  pages: PropTypes.arrayOf(PropTypes.number).isRequired,
  setPage: PropTypes.func.isRequired
};
```
Note I'm also stealing a trick from [this StackOverflow answer](https://stackoverflow.com/a/40788571) to comma-separate the page numbers and avoid including whitespace in the links, giving:

> [1](example.com), [2](example.com), [3](example.com)

instead of 
> [ 1](example.com)[ 2](example.com)[ 3](example.com)

Reply:
Fixed.